blog/docs/articles/getting-started-blog-p3.md

```diff
@@ -0,0 +1,152 @@
+
+Now that we have an understanding of how the struct works, let’s make our actual changes. Try to add a message template field into the sample source so that users can configure that message in each event. Once you’ve given this a try yourself, continue reading to see how we did it! In general, try to attempt each coding step yourself before reading the “solutions”. We will remind you about this as you continue through the blog posts.
+
+![](/blog/images/getting-started-blog-series/post3/006.png)
+
+```go
+// SampleSourceSpec holds the desired state of the SampleSource (from the client).
+type SampleSourceSpec struct {
+    // inherits duck/v1 SourceSpec, which currently provides:
+    // * Sink - a reference to an object that will resolve to a domain name or
+    //   a URI directly to use as the sink.
+    // * CloudEventOverrides - defines overrides to control the output format
+    //   and modifications of the event sent to the sink.
+    duckv1.SourceSpec `json:",inline"`
+
+    // ServiceAccountName holds the name of the Kubernetes service account
+    // as which the underlying K8s resources should be run. If unspecified
+    // this will default to the "default" service account for the namespace
+    // in which the SampleSource exists.
+    // +optional
+    ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+    // Interval is the time interval between events.
+    //
+    // The string format is a sequence of decimal numbers, each with optional
+    // fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time
+    // units are "ns", "us" (or "µs"), "ms", "s", "m", "h". If unspecified
+    // this will default to "10s".
+    Interval string `json:"interval"`
+
+    // MessageTemplate is the text/template which is used for every event sent.
+    //
+    // The string format is any valid go text/template template string where
+    // the fields are any variables set in the config map
+    MessageTemplate string `json:"messageTemplate"`
+}
+
+```
+
+1. All we added was the MessageTemplate string to the spec. Note the JSON tag - this is very important! Without a JSON tag, this field will not be read out of a JSON object when the controller receives info about it from the API Server, and the value will not be written to the JSON we send back to the API Server, so it would not be stored in etcd.
```

Comment:
nit: let's not use numbers for these sections, since we already have the numbers in the images. Also, the images break up the list formatting so only the first one is formatted properly
```suggestion
All we added was the MessageTemplate string to the spec. Note the JSON tag - this is very important! Without a JSON tag, this field will not be read out of a JSON object when the controller receives info about it from the API Server, and the value will not be written to the JSON we send back to the API Server, so it would not be stored in etcd.
```